frontend/components/SetupPage/SetupYourAgent/MemeooorrAgentForm/index.tsx

```diff
@@ -0,0 +1,305 @@
+import { EyeInvisibleOutlined, EyeTwoTone } from '@ant-design/icons';
+import { Button, Divider, Flex, Form, Input, message, Typography } from 'antd';
+import React, { useCallback, useState } from 'react';
+import { useUnmount } from 'usehooks-ts';
+
+import { ServiceTemplate } from '@/client';
+import { CustomAlert } from '@/components/Alert';
+import { SetupScreen } from '@/enums/SetupScreen';
+import { useElectronApi } from '@/hooks/useElectronApi';
+import { useSetup } from '@/hooks/useSetup';
+import { useStakingProgram } from '@/hooks/useStakingProgram';
+
+import { onDummyServiceCreation } from '../utils';
+import { validateGeminiApiKey, validateTwitterCredentials } from './validation';
+
+const { Title, Text } = Typography;
+
+type FieldValues = {
+  personaDescription: string;
+  geminiApiKey: string;
+  xEmail: string;
+  xUsername: string;
+  xPassword: string;
+};
+type ValidationStatus = 'valid' | 'invalid' | 'unknown';
+
+const requiredRules = [{ required: true, message: 'Field is required' }];
+const validateMessages = {
+  required: 'Field is required',
+  types: { email: 'Enter a valid email' },
+};
+const commonFieldProps = { rules: requiredRules, hasFeedback: true };
```

Comment:
wonder if we could move this to `SetupYourAgent/utils.ts` or create `SetupYourAgent/constants.ts` or even `SetupYourAgent/validations.ts`, this is partially repeated in ModiusAgentForm